### PR TITLE
optimize: do layer filesystem initialization on start

### PR DESCRIPTION
**Pull request by justadogistaken**
There is a new package named onstart, we can do some initialization job within the package.

### CODE REVIEW

File: sealer/onstart/onstart.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package onstart
```
Comment (fanux) [inline sealer/onstart/onstart.go:1]: The name of the module is a little strange.

Comment (fanux) [reply sealer/onstart/onstart.go:1]: How about call this module `boot`?

File: image/store/filesystem.go [hunk 2]
```diff
@@ -172,6 +172,16 @@ func (ls LayerStorage) assembleTar(id LayerID) (io.ReadCloser, error) {
 	}), nil
 }
 
+func InitLayerDirectory() error {
+	dls := NewDefaultLayerStorage()
```
Comment (touchuyht) [inline image/store/filesystem.go:176]: I wonder if it is necessary to call this function here. We could just use layerDataRoot and layerDBRoot.

File: sealer/onstart/onstart.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+	err = utils.MkDirIfNotExists(common.DefaultImageMetaRootDir)
+	if err != nil {
+		return fmt.Errorf("failed to mk sealer metadata root dir, err: %s", err)
+	}
+	return nil
+}
+
+func OnBoot() error {
+	err := initRootDirectory()
```
Comment (touchuyht) [inline sealer/onstart/onstart.go:26]: The actions of theses functions are similar, maybe we could create an array which saves names of the files to be created and we range the array to create them.

**Comment by justadogistaken** [comment]
@fanux @touchuyht PTAL

**Review by fanux — APPROVED** [review]

**Pull request merged**